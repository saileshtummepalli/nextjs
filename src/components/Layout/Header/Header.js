import React from "react";
import Link from "next/link"
function Header(props) {
  return (
    <div data-test="HeaderComponent">
      <div
        className=" py-20 from-blue-500 to-purple-400 bg-gradient-to-r w-full h-64 block">
        <div className="container mx-auto px-6">
          <h2 className=" uppercase font-bold text-2xl text-black">{props.hello} </h2>
          <h3 className="text-2xl mb-8 text-gray-200">{props.letsgo}</h3>
          {props.button && <Link href="/dogs">
          <button className="bg-white font-bold rounded py-4 px-8 shadow-lg uppercase tracking-wider">wooof</button>
          </Link>}
        </div>
      </div>
    </div>
  );
}

export default Header;
