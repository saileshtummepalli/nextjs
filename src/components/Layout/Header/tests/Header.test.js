import React from 'react';
import { shallow} from "enzyme"
import  Header  from '../Header';

  describe('<Header/>', () => {
    it('Should shallow correctly', () => {
      const  container  = shallow( <Header letsgo="test" hello="hello" />);
      expect(container.debug()).toMatchSnapshot();
    });
    describe('<Header/>', () => {
      it('enzyme render', () => {
        const  container  = shallow(<Header/>);
        const wapper = container.find(`[data-test='HeaderComponent']`);
        expect(wapper.length).toBe(1);
      });
    })
  });