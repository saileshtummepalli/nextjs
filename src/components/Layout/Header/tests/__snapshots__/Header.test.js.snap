// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Should shallow correctly 1`] = `
"<div data-test=\\"HeaderComponent\\">
  <div className=\\" py-20 from-blue-500 to-purple-400 bg-gradient-to-r w-full h-64 block\\">
    <div className=\\"container mx-auto px-6\\">
      <h2 className=\\" uppercase font-bold text-2xl text-black\\">
        hello
         
      </h2>
      <h3 className=\\"text-2xl mb-8 text-gray-200\\">
        test
      </h3>
    </div>
  </div>
</div>"
`;
